### PR TITLE
[TP] Add API logging for TP high level API

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102209

### CODE REVIEW

**Review by wz337 — APPROVED** [review]
LGTM

File: torch/distributed/tensor/parallel/fsdp.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ def enable_2d_with_fsdp() -> bool:
     Return:
         A `bool` indicated whether extension registration succeeds or not.
     """
+
+    torch._C._log_api_usage_once("torch.distributed.tensor.parallel.enable_2d_with_fsdp")
```
Comment (wanchaol) [inline torch/distributed/tensor/parallel/fsdp.py:47]: is this indention global? I think it should be local to `enable_2d_with_fsdp` function itself?

File: torch/distributed/tensor/parallel/api.py [hunk 2]
```diff
@@ -80,6 +80,8 @@ def parallelize_module(  # type: ignore[return]
         granularity, you need to pass in a dict of module FQN and parallel style instead.
     """
 
+    torch._C._log_api_usage_once("torch.distributed.tensor.parallel.parallelize_module")
```
Comment (wanchaol) [inline torch/distributed/tensor/parallel/api.py:83]: same question: I think it should only be called inside the API instead of in a global scope?

Comment (fduwjj) [reply torch/distributed/tensor/parallel/api.py:83]: Looks like they are 4 spaces same as L85. Or unless I am missing anything here?

File: torch/distributed/tensor/parallel/fsdp.py (hunk 1 above)
Comment (fduwjj) [reply torch/distributed/tensor/parallel/fsdp.py:47]: The indention is same as L49.

File: torch/distributed/tensor/parallel/api.py (hunk 2 above)
Comment (wanchaol) [reply torch/distributed/tensor/parallel/api.py:83]: hmmm nvm, looks like github UI shows it's a different indention when my screen is small.

**Review by wanchaol — APPROVED** [review]
lgtm

**Pull request closed without merging**